### ISSUE TITLE
Cannot search in Chinese

### ISSUE THREAD

**Issue by A3148475824** [kiwix/kiwix-android#3889]
English can be searched for words or vocabulary! But there is only one child in Chinese, and it cannot be searched! Search is broken

**Comment by A3148475824** [comment]
Bookmarks can only show 25

**Comment by kelson42** [comment]
@A3148475824 Thank you formyour report. The problem with Chinese character is known and is on the scraper side, see https://github.com/openzim/mwoffliner/issues/1576.

For the bookmark problem, not sure to understand what you mean. Counyounplease open a dedicated issue with the exact steps to follow to reproduce the problem?

**Comment by A3148475824** [comment]
> @A3148475824 Thank you formyour report. The problem with Chinese character is known and is on the scraper side, see [openzim/mwoffliner#1576](https://github.com/openzim/mwoffliner/issues/1576).
> 
> For the bookmark problem, not sure to understand what you mean. Counyounplease open a dedicated issue with the exact steps to follow to reproduce the problem?

I don't know, maybe it's a bug. When I checked at first, it showed only 25 bookmarks! But there are definitely more than 50. I restarted the phone and it didn't fix it! But now I opened it and found that it has been fixed again, and all the bookmarks have appeared! I don't quite understand this bug, and I can't reproduce it

**Comment by kelson42** [comment]
@MohitMaliDeveloper Can you please check we can store/display more than 25 bookmarks?

**Comment by A3148475824** [comment]
> @MohitMaliDeveloper Can you please check we can store/display more than 25 bookmarks?

![Screenshot_2010-01-01-01-00-58](https://github.com/kiwix/kiwix-android/assets/70214970/f4732949-ff5b-4f81-a7d3-2d661137c4b4)
![Screenshot_2010-01-01-01-00-52](https://github.com/kiwix/kiwix-android/assets/70214970/5d64b07f-2476-45f4-acdc-14c80b41ab33)

I can only tell you through pictures that some of the bookmarks disappeared and need to be searched to appear. The specific number is not important. What is important is that the bookmarks will disappear! After all, I don’t know how many bookmarks I have collected!

**Comment by A3148475824** [comment]
![Uploading Screenshot_20240624-210949_Kiwix.png…]()

**Comment by A3148475824** [comment]
![Uploading b85aee3a5baf4e259d6ab46fb8e029ca.jpeg…]()

**Comment by A3148475824** [comment]
> @MohitMaliDeveloper Can you please check we can store/display more than 25 bookmarks?

I can only tell you through pictures that some of the bookmarks disappeared and need to be searched to appear. The specific number is not important. What is important is that the bookmarks will disappear! After all, I don’t know how many bookmarks I have collected!
I don't know how to explain it! I opened bookmark.xml, it's in /storage/emulated/0/Android/data/org.kiwix.kiwixmobile/files/Bookmarks! I opened it and counted 43 bookmarks! I also counted 43 bookmarks in kiwix bookmarks! But this is a bug, because I have many bookmarks, far more than these. I used the bookmark search function in kiwix to search for Hong Kong, but it showed more than a dozen bookmarks, proving that I have already collected it, but the bookmark function did not show it.



![Screenshot_20240624-211023_Kiwix](https://github.com/kiwix/kiwix-android/assets/70214970/4b0af991-0315-4c9a-bc07-16a5fb4c5e70)
![b85aee3a5baf4e259d6ab46fb8e029ca](https://github.com/kiwix/kiwix-android/assets/70214970/634b0ef1-6dbc-4b04-8811-d5d0203f28d0)

**Comment by MohitMaliFtechiz** [comment]
> @MohitMaliDeveloper Can you please check we can store/display more than 25 bookmarks?

@kelson42 Yes, we can store/display more than 25 bookmarks.

@A3148475824 Thanks for describing the behavior, it was very helpful in debugging the problem :innocent:. We have fixed this in https://github.com/kiwix/kiwix-android/pull/3893.

**Comment by A3148475824** [comment]
> > @MohitMaliDeveloper Can you please check we can store/display more than 25 bookmarks?
> 
> @kelson42 Yes, we can store/display more than 25 bookmarks.
> 
> @A3148475824 Thanks for describing the behavior, it was very helpful in debugging the problem 😇. We have fixed this in #3893.

If it can be packaged into apk and sent to me and let me test whether it can run, I think it can solve the problem more quickly, or a new unknown error occurs!

**Comment by MohitMaliFtechiz** [comment]
@A3148475824 You can test this APK https://drive.google.com/file/d/11JYfBfYf3H4dIUibZ0tuMlT_vtOEm0xn/view?usp=drive_link

**Comment by A3148475824** [comment]
> @A3148475824 You can test this APK https://drive.google.com/file/d/11JYfBfYf3H4dIUibZ0tuMlT_vtOEm0xn/view?usp=drive_link

Hello, after my test, I found a few problems! The first time you enter the software and click the menu button, there will be a lag. Of course, it will only be the first time you click the menu button, and it will not be there after that. But if you click the settings button every time, red lines will appear around the screen for a short time! And you still can't search in Chinese! But the problem of bookmarks has been solved, and there will be no non-display situation
